File: backend/internal/model/chain/chain.go
package chain

type Chain string

const (
	Solana Chain = "solana"
	Sui    Chain = "sui"
)

var Supported = []Chain{Solana, Sui}

func (c Chain) IsSupported() bool {
	for _, s := range Supported {
		if c == s {
			return true
		}
	}
	return false
}

File: mobile/chains/chain.ts
//Chain - all chains that are supported by the app
export const Chain = {
    Solana: 'solana',
    Sui: 'sui',
} as const;

export type Chain = (typeof Chain)[keyof typeof Chain];
export const AllChains = Object.values(Chain);

export const ChainColors = {
    [Chain.Solana]: '#D724FD',
    [Chain.Sui]: '#4DA2FF',
};
